modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java

```diff
@@ -585,117 +732,64 @@ private void storeEntries(Collection<IgniteBiTuple<ByteArray, byte[]>> entries,
             return deployFut.thenApply(uid -> id);
     }
 
-    /**
-     * Checks whether the given node hosts meta storage.
-     *
-     * @param nodeName Node unique name.
-     * @param metastorageMembers Meta storage members names.
-     * @return {@code true} if the node has meta storage, {@code false} otherwise.
-     */
-    public static boolean hasMetastorage(String nodeName, String[] metastorageMembers) {
-        boolean isNodeHasMetasorage = false;
-
-        for (String name : metastorageMembers) {
-            if (name.equals(nodeName)) {
-                isNodeHasMetasorage = true;
-
-                break;
-            }
-        }
-
-        return isNodeHasMetasorage;
-    }
-
     /**
      * Checks whether the local node hosts meta storage.
      *
-     * @param configurationMgr Configuration manager.
      * @return {@code true} if the node has meta storage, {@code false} otherwise.
      */
-    public boolean hasMetastorageLocally(ConfigurationManager configurationMgr) {
-        String[] metastorageMembers = configurationMgr
-            .configurationRegistry()
-            .getConfiguration(NodeConfiguration.KEY)
-            .metastorageNodes()
-            .value();
-
-        return hasMetastorage(vaultMgr.name().join(), metastorageMembers);
+    public boolean hasMetastorageLocally() {
+        return metaStorageNodeNamesFut
+            .thenCompose(metastorageNodes ->
+                vaultMgr.name().thenApply(name -> metastorageNodes.stream().anyMatch(name::equals))
```

Comment:
Should we get the name from topologyService.localMember() instead?

Reply:
I don't know, this logic was already present here. But looks like you are correct....